frontend/src/stores/QuestionStore.ts

```diff
@@ -99,16 +99,31 @@ export const useQuestionStore = defineStore('question', () => {
   }
 
   function revertAnswer(previousSubCategory: string) {
+    /**
+     * This function reverts the answer to the question if the user goes back a question.
+     * For this purpose the QuestionId, answers, labels for the subcategory are reverted.
+     * If there is no label for the subcategory, "nader te bepalen" is assigned.
+     */
     QuestionId.value = answers.value[answers.value.length - 1]
     answers.value.pop()
+
     if (labels.value[QuestionId.value]) {
-      const label: string = labels.value[QuestionId.value]
-      LabelsBySubCategory.value[previousSubCategory].pop(label)
+      const questionLabels: string[] = labels.value[QuestionId.value]
+
+      questionLabels.forEach(label => {
+        const labelIndex = LabelsBySubCategory.value[previousSubCategory].indexOf(label)
+        if (labelIndex !== -1) {
+          LabelsBySubCategory.value[previousSubCategory].splice(labelIndex, 1)
+        }
+      })
+
       if (LabelsBySubCategory.value[previousSubCategory].length === 0) {
-        LabelsBySubCategory.value[previousSubCategory].push('nader te bepalen')
+        LabelsBySubCategory.value[previousSubCategory] = ['nader te bepalen']
```

Comment:
```suggestion
        LabelsBySubCategory.value[previousSubCategory] = ['nader te bepalen']
```
Wouldn't this still be push but then instead of pushing the string you push the list of the string? As this is how the sessionstorage expects it?

Reply:
LabelsBySubCategory.value[previousSubCategory].push(['nader te bepalen']) so like this?